packages/cli/src/controllers/me.controller.ts

```diff
@@ -46,10 +51,34 @@ export class MeController {
 		res: Response,
 		@Body payload: UserUpdateRequestDto,
 	): Promise<PublicUser> {
-		const { id: userId, email: currentEmail, mfaEnabled } = req.user;
-
-		const { email } = payload;
+		const {
+			id: userId,
+			email: currentEmail,
+			mfaEnabled,
+			firstName: currentFirstName,
+			lastName: currentLastName,
+		} = req.user;
+
+		const { email, firstName, lastName } = payload;
 		const isEmailBeingChanged = email !== currentEmail;
+		const isFirstNameChanged = firstName !== currentFirstName;
```

Comment:
(completely optional, just if you think it makes sense)
Maybe we could have some kind of map of auth method => check if can change profile, in order to merge both ldap / oids and saml check into one (because it's doing the same business logic)